File: IT/Programmazione funzionale/Python/part.py
# file part.py

from __future__ import print_function, division
from functools import partial

# funzione che calcola la percentuale, ad es. il 60% di 120

percent = lambda x, y : x / 100 * y

# calcola il 60% di 120: invoca la funzione con entrambi i parametri

print("60% di 120 = {}".format(percent(120, 60)))

# applicazione parziale della funzione: fissa il primo parametro al variare del secondo
part = partial(percent, 70)

print("30% di 70 = {:.1f}".format(part(30)))
print("60% di 70 = {:.1f}".format(part(60)))
print("90% di 70 = {:.1f}".format(part(90)))

File: IT/2/Programmazione funzionale/C++/part.cpp
/*
file part.cpp
compilare con: g++ -std=c++11 part.cpp
*/

#include <iostream>
#include <functional>

using std::cout;
using std::endl;
using std::bind;
using namespace std::placeholders;

int main() {
	// funzione che calcola la percentuale, ad es. il 30% di 100
	auto percent = [] (const int x, const int y) { return (double) x / 100 * y; };

	// calcola il 60% di 120: invoca la funzione con entrambi i parametri
	cout << "60% di 120 = " << percent(120, 60) << endl;

	// applicazione parziale della funzione: fissa il primo parametro al variare del secondo

	auto part = bind(percent, 70, _1);

	cout << "30% di 70 = " << part(30) << endl;
	cout << "60% di 70 = " << part(60) << endl;
	cout << "90% di 70 = " << part(90) << endl;
}

